lib/bootloader_setup.pm

```diff
@@ -56,6 +58,91 @@ sub stop_grub_timeout {
     send_key 'up';
 }
 
+=head2 add_custom_grub_entries
+
+Add custom grub entries with extra kernel parameters.
+It adds 3rd line with default options + 4th line with advanced options.
+Extra kernel parameters are taken in GRUB_PARAM variable.
+
+e.g.  grub entries before:
+* SLES 15
+* Advanced options for SLES 15
+* Start bootloader from a read-only snapshot
+
+grub entries with GRUB_PARAM='ima_policy=tcb' and calling add_custom_grub_entries:
+* SLES 15
+* Advanced options for SLES 15
+* SLES 15 (ima_policy=tcb)
+* Advanced options for SLES 15 (ima_policy=tcb)
+* Start bootloader from a read-only snapshot
+
+And of course the new entries have ima_policy=tcb added to kernel parameters.
+
+=cut
+sub add_custom_grub_entries {
```

Comment:
The current file lib/bootloader_setup.pm is already more than 700 lines long which is really hard to maintain and also this function does not quite fit what other functions are doing. All the others if I am not mistaken handle the boot menu in place, this one updates grub files within the booted system before rebooting into a new system. I guess we can find a better place

Reply:
Well it does bootloader setup, as grub is a boot loader :). And this file is handling grub, so it perfectly fits to it. I'll create `lib/grub_utils.pm` if you want, but IMHO it's confusing. 
And I'd like to avoid doing cleanup with deciding what should be in this new module and what should not. If there is a need of cleanup bootloader setup, it should be done separately and not blocking fixing IMA tests, which has been waiting long time to be fixed.

Reply:
Ok, I was not aware that any IMA tests need urgent fixing. Your changes looked more like "Extend test coverage" which should normally be handled with a bit more care to not introduce new problems :) Ok, leave it for the time being :)